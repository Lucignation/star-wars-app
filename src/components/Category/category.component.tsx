import { FC } from 'react';
import { ICategory } from '../../common/interfaces/interface';
import styles from './category.module.scss';

type props = {
  category: ICategory;
};
const Category: FC<props> = ({ category }) => {
  const bg = {
    content: '',
    backgroundColor: category.colorCode,
    borderRadius: 5 + `px`,
    width: 27 + `px`,
    height: 26 + `px`,
  };
  return (
    <div className={styles.category_container}>
      <div className={styles.category_top_info}>
        <h2 className={styles.category_title}>{category.title}</h2>
        <p style={bg}></p>
      </div>
      <div className={styles.category_total}>{category.total}</div>
      <p className={styles.category_info}>{category.info}</p>
    </div>
  );
};

export default Category;
