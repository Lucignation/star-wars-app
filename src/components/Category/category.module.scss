.category_container {
  background: #ffffff;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 10px;
  padding: 24px 18px;

  //   width: 208px;
  //   height: 130px;
}

.category_top_info {
  display: grid;
  grid-template-columns: 80% 20%;
}

.category_title {
  font-family: 'Inter', sans-serif;
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
}

.category_total {
  font-family: 'Inter', sans-serif;
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
  margin-top: 23px;
}

.category_info {
  font-family: 'Inter', sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 9px;
  line-height: 24px;
  /* identical to box height, or 267% */

  color: #00992b;
  margin-top: -7px;
}
