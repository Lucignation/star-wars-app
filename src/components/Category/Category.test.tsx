import Category from './category.component';
import renderer from 'react-test-renderer';

const _category = {
  title: 'title',
  total: 30,
  info: 'category information',
  colorCode: '#fff',
};

const mockProps = {
  category: _category,
};

describe('Category', () => {
  it('enders a snapshot of <Category /> when passed into a component', () => {
    const category = renderer.create(<Category {...mockProps} />).toJSON();

    expect(category).toMatchSnapshot();
  });
});
