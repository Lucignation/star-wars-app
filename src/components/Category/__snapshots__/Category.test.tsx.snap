// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category enders a snapshot of <Category /> when passed into a component 1`] = `
<div
  className="category_container"
>
  <div
    className="category_top_info"
  >
    <h2
      className="category_title"
    >
      title
    </h2>
    <p
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": "5px",
          "content": "",
          "height": "26px",
          "width": "27px",
        }
      }
    />
  </div>
  <div
    className="category_total"
  >
    30
  </div>
  <p
    className="category_info"
  >
    category information
  </p>
</div>
`;
